Source/geom.c

```diff
@@ -259,3 +260,22 @@ void tesedgeIntersect( TESSvertex *o1, TESSvertex *d1,
 		v->t = Interpolate( z1, o2->t, z2, d2->t );
 	}
 }
+
+/*
+	Calculate the angle between v1-v2 and v1-v0
+ */
+TESSreal calcAngle(TESSvertex *v0, TESSvertex *v1, TESSvertex *v2) {
+	TESSreal a[2] = { v2->s - v1->s, v2->t - v1->t };
+	TESSreal b[2] = { v0->s - v1->s, v0->t - v1->t };
+	return acosf((a[0] * b[0] + a[1] * b[1]) /

```

Comment:
Should use `acos` instead of `acosf`. acos has tendency to be picky about inputs, so it would make sense to clamp it. Also, should guard against div/0.
```
TESSreal num = (a[0] * b[0] + a[1] * b[1]);
TESSreal den = sqrt( a[0] * a[0] + a[1] * a[1] ) * sqrt( b[0] * b[0] + b[1] * b[1]) );
if ( den > 0.0 )
	num /= den;
if ( num < -1.0 ) num = -1.0;
if ( num >  1.0 ) num =  1.0;
return acos( num );
```